### PR TITLE
fix(inspector): add back list endpoint

### LINKED ISSUE #8870: "Promise was canceled" error when debugging in v1.6.2
It looks like a debugger regression was introduced in v1.6.2. I can debug from VS Code fine with v1.6.1 and earlier using a configuration like this:
```
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Debug Deno",
      "type": "pwa-node",
      "request": "launch",
      "cwd": "${workspaceFolder}",
      "runtimeExecutable": "deno",
      "program": "${file}",
      "runtimeArgs": [
        "run",
        "--inspect-brk",
        "--allow-all"
      ],
      "outputCapture": "std",
      "attachSimplePort": 9229
    }
  ]
}
```
However, upgrading to v1.6.2 results in this error when trying to connect the debugger:
```
Error processing launch: Error: Could not connect to debug target at http://127.0.0.1:9229: Promise was canceled
```
A temporary fix is to just roll back to the last working version:
```
deno upgrade --version 1.6.1
```

### PR DESCRIPTION
**Pull request by lperlaki**
fix #8870

### CODE REVIEW

**Comment by caspervonb** [comment]
This is a regression from the warp removal?

File: runtime/inspector.rs [hunk 1]
```diff
@@ -275,11 +276,14 @@ async fn server(
...
+            (&http::Method::GET, "/json/version") => {
+              handle_json_version_request(json_version_response.clone())
+            }
             (&http::Method::GET, "/json") => {
               handle_json_request(inspector_map.clone())
             }
-            (&http::Method::GET, "/json/version") => {
-              handle_json_version_request(json_version_response.clone())
+            (&http::Method::GET, "/json/list") => {
+              handle_json_request(inspector_map.clone())
```
Comment (bartlomieju) [inline runtime/inspector.rs:286]: Looks like this route has been inadvertently removed when rewriting inspector server. @lperlaki please add a test case in `cli/tests/integration_tests.rs` that hits this endpoint

**Review by bartlomieju — CHANGES_REQUESTED** [review]

File: runtime/inspector.rs (hunk 1 above)
Comment (lperlaki) [reply runtime/inspector.rs:286]: @bartlomieju I wil do that but I couldn't find how any other integration test that did http request, what library/method should I use?

Comment (bartlomieju) [reply runtime/inspector.rs:286]: @lperlaki you can use `reqwest` for that purpose

File: cli/tests/integration_tests.rs [hunk 2]
```diff
@@ -4333,6 +4333,66 @@ async fn inspector_runtime_evaluate_does_not_crash() {
...
+
+  let stderr = child.stderr.as_mut().unwrap();
+  let mut stderr_lines =
+    std::io::BufReader::new(stderr).lines().map(|r| r.unwrap());
+  let ws_url = extract_ws_url_from_stderr(&mut stderr_lines);
+  let mut url = ws_url.clone();
+  let _ = url.set_scheme("http");
+  url.set_path("/json");
+  let resp = reqwest::get(url).await.unwrap();
+  assert_eq!(resp.status(), reqwest::StatusCode::from_u16(200).unwrap()); //
```
Comment (bnoordhuis) [inline cli/tests/integration_tests.rs:4355]:
```suggestion
  assert_eq!(resp.status(), reqwest::StatusCode::OK);
```

File: cli/tests/integration_tests.rs [hunk 3]
```diff
@@ -4333,6 +4333,66 @@ async fn inspector_runtime_evaluate_does_not_crash() {
...
+  let enpoint_list = resp
+    .json::<Vec<deno_core::serde_json::Value>>()
+    .await
+    .unwrap();
+  let matching_endpoint = enpoint_list
```
Comment (bnoordhuis) [inline cli/tests/integration_tests.rs:4360]:
```suggestion
  let endpoint_list = resp
    .json::<Vec<deno_core::serde_json::Value>>()
    .await
    .unwrap();
  let matching_endpoint = endpoint_list
```

File: cli/tests/integration_tests.rs [hunk 4]
```diff
@@ -4333,6 +4333,66 @@ async fn inspector_runtime_evaluate_does_not_crash() {
...
+
+  let stderr = child.stderr.as_mut().unwrap();
+  let mut stderr_lines =
+    std::io::BufReader::new(stderr).lines().map(|r| r.unwrap());
+  let ws_url = extract_ws_url_from_stderr(&mut stderr_lines);
+  let mut url = ws_url.clone();
+  let _ = url.set_scheme("http");
+  url.set_path("/json/list");
+  let resp = reqwest::get(url).await.unwrap();
+  assert_eq!(resp.status(), reqwest::StatusCode::from_u16(200).unwrap()); //
```
Comment (bnoordhuis) [inline cli/tests/integration_tests.rs:4385]:
```suggestion
  assert_eq!(resp.status(), reqwest::StatusCode::OK);
```

File: cli/tests/integration_tests.rs [hunk 5]
```diff
@@ -4333,6 +4333,66 @@ async fn inspector_runtime_evaluate_does_not_crash() {
...
+  let enpoint_list = resp
+    .json::<Vec<deno_core::serde_json::Value>>()
+    .await
+    .unwrap();
+  let matching_endpoint = enpoint_list
```
Comment (bnoordhuis) [inline cli/tests/integration_tests.rs:4390]:
```suggestion
  let endpoint_list = resp
    .json::<Vec<deno_core::serde_json::Value>>()
    .await
    .unwrap();
  let matching_endpoint = endpoint_list
```

**Review by bnoordhuis — APPROVED** [review]
Minor nits but otherwise LGTM. Thanks for fixing this!

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks @lperlaki

**Pull request merged**